EdiEngine/Validation/DataElementValidator.cs

```diff
@@ -48,33 +44,100 @@ public static bool IsValid(this EdiSimpleDataElement el, MapSimpleDataElement de
-                        (definition.ReqDes == RequirementDesignator.Optional && string.IsNullOrEmpty(el.Val));
-
+                    switch (definition.MaxLength)
+                    {
+                        case 4:
+                            return ValidateShortTime(el, definition);
+                        case 8:
+                            return ValidateLongTime(el, definition);
+                        default:
+                            throw new NotSupportedException($"Time validation with length {definition.MaxLength} is not implemented. {el.Type}");
+                    }
                 default:
                     throw new NotSupportedException($"{definition.DataType} validation is not implemented. {el.Type}");
             }
         }
+
+        #region PrivateHelpers
+
+        private static bool ValidateLongDate(EdiSimpleDataElement el, MapSimpleDataElement definition)
+        {
+            if (string.IsNullOrEmpty(el.Val))
+                return IsOptional(el.Val, definition);
+
+            return el.Val.IsValidDateTimeString("yyyyMMdd");
+        }
+
+        private static bool ValidateShortDate(EdiSimpleDataElement el, MapSimpleDataElement definition)
+        {
+            if (string.IsNullOrEmpty(el.Val))
+                return IsOptional(el.Val, definition);
+
+            return el.Val.IsValidDateTimeString("yyMMdd");
+        }
+
+        private static bool ValidateShortTime(EdiSimpleDataElement el, MapSimpleDataElement definition)
+        {
+            if (string.IsNullOrEmpty(el.Val))
+                return IsOptional(el.Val, definition);
+
+            switch (el.Val.Length)
+            {
+                case 4:
+                    return el.Val.IsValidDateTimeString("HHmm");
+
+                default:
+                    throw new NotSupportedException($"Time validation with length {el.Val.Length} is not implemented. {el.Type}");
+            }
+        }
+
+        private static bool ValidateLongTime(EdiSimpleDataElement el, MapSimpleDataElement definition)
+        {
+            if (string.IsNullOrEmpty(el.Val))
+                return IsOptional(el.Val, definition);
+
+            switch (el.Val.Length)
+            {
+                case 4:
+                    return ValidateShortTime(el, definition);
+                case 5:
+                    return el.Val.IsValidDateTimeString("HHmms");
+                case 6:
+                    return el.Val.IsValidDateTimeString("HHmmss");
+                case 7:
+                    return el.Val.IsValidDateTimeString("HHmmssf");
+                case 8:
+                    return el.Val.IsValidDateTimeString("HHmmssff");
+                default:
+                    throw new NotSupportedException($"Time validation with length {el.Val.Length} is not implemented. {el.Type}");
+            }
+        }
+
+        private static bool IsValidDateTimeString(this string dateTime, string format)
+        {
```

Comment:
Hi Sharif, 
Can you change it so it compiles in C# 6 and VS 2015 - as i am maintaining compatibility.

        private static bool IsValidDateTimeString(this string dateTime, string format)
        {
           DateTime dummyDt;
            return DateTime.TryParseExact(dateTime, format, CultureInfo.InvariantCulture, DateTimeStyles.None, out dummyDt);
        }

Reply:
Fixed!